uber/site_sections/registration.py

```diff
@@ -82,6 +85,18 @@ def index(self, session, message='', page='0', search_text='', uploaded_id='', o
     def form(self, session, message='', return_to='', omit_badge='', **params):
         attendee = session.attendee(params, checkgroups=['interests','requested_depts','assigned_depts'],
                                     bools=['staffing','trusted','international','placeholder','got_merch','can_spam'])
+
+        if 'watchlist_id' in params:
+            watchlist_entry = session.watch_list(params['watchlist_id'])
+            watchlist_entry.disabled = True
+            session.add(watchlist_entry)
+            session.commit() # Need to commit now or the attendee will be automatically set to Deferred
+
+            if attendee.status == DEFERRED_STATUS:
+                attendee.status = NEW_STATUS
+            session.add(attendee)
```

Comment:
Hm, actually we really shouldn't be assuming that there will always be an attendee here. For example, we may want to later let admins create attendees and add them to the watch list that way (IMO this would be nicer than trying to match names). We don't want it to explode when we make that change on the front-end.